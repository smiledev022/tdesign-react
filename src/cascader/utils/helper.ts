import { TreeNode } from '../interface';

/**
 * 获取node节点的全路径
 * @param node
 * @returns
 */
export function getFullPathLabel(node: TreeNode) {
  return node
    .getPath()
    .map((node: TreeNode) => node.label)
    .join('/');
}
