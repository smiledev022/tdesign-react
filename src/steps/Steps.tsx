import React, { useEffect, useRef } from 'react';
import classnames from 'classnames';
import useConfig from '../_util/useConfig';
import Step from './Step';
import StepsContext from './StepsContext';
import { StepsProps } from './StepsProps';

/**
 * 步骤条组件
 * @param props
 */
export default function Steps(props: StepsProps) {
  const {
    style,
    current = 1,
    direction = 'horizontal',
    status,
    type = 'default',
    sequence = 'positive',
    children,
    onChange,
  } = props;
  const { classPrefix } = useConfig();

  const className = classnames({
    [`${classPrefix}-steps`]: true,
    [`${classPrefix}-steps--horizontal`]: direction === 'horizontal',
    [`${classPrefix}-steps--vertical`]: direction === 'vertical',
    [`${classPrefix}-steps--default-anchor`]: type === 'default',
    [`${classPrefix}-steps--positive`]: sequence === 'positive',
    [`${classPrefix}-steps--reverse`]: sequence === 'reverse',
    [`${classPrefix}-steps--dot-anchor`]: type === 'dot',
    [props.className]: !!props.className,
  });

  const previousRef = useRef<number>(current);

  // 监听步骤变化
  useEffect(() => {
    const previous = previousRef.current;
    if (previous !== current && onChange) {
      onChange(current, previous);
    }
  }, [current, onChange]);

  return (
    <StepsContext.Provider value={{ current, currentStatus: status, type }}>
      <div className={className} style={style}>
        {React.Children.map(children, (child: JSX.Element, index: number) => {
          let stepNumber = index + 1;
          // 垂直状态下、反序
          if (sequence === 'reverse' && direction === 'vertical') {
            const childs = children as any;
            stepNumber = childs.length - index;
          }
          return React.cloneElement(child, { stepNumber });
        })}
      </div>
    </StepsContext.Provider>
  );
}

Steps.Step = Step;
Steps.displayName = 'Steps';
