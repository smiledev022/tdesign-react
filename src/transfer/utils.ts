import isEmpty from 'lodash/isEmpty';
import isFunction from 'lodash/isFunction';
import isString from 'lodash/isString';
import React from 'react';
import { TNode } from '../common';

export const getDefaultValue = <T>(value: T | T[], defaultValue?: T): T[] => {
  if (isEmpty(value)) {
    return Array.isArray(defaultValue) ? defaultValue : [defaultValue, defaultValue];
  }

  if (Array.isArray(value)) return value;

  return [defaultValue, defaultValue];
};

export const getJSX = (value: string | TNode) => {
  if (isString(value)) return value;

  if (isFunction(value)) return value();

  if (React.isValidElement(value)) return value;

  return value;
};
