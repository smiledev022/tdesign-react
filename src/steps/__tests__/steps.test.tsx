import React from 'react';
import { testExamples, render } from '@test/utils';
import Steps from '../Steps';

const { StepItem } = Steps;

// 测试组件代码 Example 快照
testExamples(__dirname);

describe('Steps 组件测试', () => {
  test('mount, unmount 测试', () => {
    const wrapper = render(
      <Steps current={1}>
        <StepItem status="finish" title="1" content="这里是提示文字"></StepItem>
        <StepItem status="process" title="2" content="这里是提示文字"></StepItem>
        <StepItem status="error" title="3" content="这里是提示文字"></StepItem>
        <StepItem status="wait" title="4" content="这里是提示文字"></StepItem>
      </Steps>,
    );

    expect(() => {
      wrapper.unmount();
    }).not.toThrow();
  });
});
