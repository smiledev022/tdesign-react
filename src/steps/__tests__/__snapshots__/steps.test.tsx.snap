// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperateSteps.jsx 1`] = `
<DocumentFragment>
  <button
    class="t-button t-button--primary t-size-s"
  >
    上一步
  </button>
  <button
    class="t-button t-button--line t-size-s"
  >
    下一步
  </button>
  <br />
  <br />
  <div
    class="t-steps t-steps--horizontal t-steps--default-anchor t-steps--positive"
  >
    <div
      class="t-steps-item t-steps-item--process"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            1
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            步骤1
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            2
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            步骤2
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            3
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            步骤3
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            4
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            步骤4
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="t-steps t-steps--vertical t-steps--default-anchor t-steps--positive"
  >
    <div
      class="t-steps-item t-steps-item--process"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            1
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            步骤1
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            2
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            步骤3
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            3
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            步骤4
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            4
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            步骤5
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ReverseSteps.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-steps t-steps--vertical t-steps--default-anchor t-steps--reverse"
  >
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            4
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            未进行的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            3
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            未进行的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--process"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            2
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            进行中的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--finish"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <svg
            class="t-icon t-icon-check-circle-filled"
            version="1.1"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM690.8655168 353.58818368L464.56401024 579.82818368 328.82684416 444.09785152 283.57201024 489.35268544 464.59134592 670.3720192 509.84617984 625.1171872 736.120352 398.8430176 690.8655168 353.58818368Z"
              fill-opacity=".9"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            已完成的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`extra.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-steps t-steps--vertical t-steps--default-anchor t-steps--positive"
  >
    <div
      class="t-steps-item t-steps-item--finish"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <svg
            class="t-icon t-icon-check-circle-filled"
            version="1.1"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM690.8655168 353.58818368L464.56401024 579.82818368 328.82684416 444.09785152 283.57201024 489.35268544 464.59134592 670.3720192 509.84617984 625.1171872 736.120352 398.8430176 690.8655168 353.58818368Z"
              fill-opacity=".9"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            已完成
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
          <div
            class="t-steps-item-extra"
          >
            <button
              class="t-button t-button--primary t-size-s"
            >
              按钮
            </button>
            <button
              class="t-button t-button--line t-size-s"
            >
              按钮
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--process"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            2
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            进行中
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            3
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            未进行
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            4
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            未进行
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`icon.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-steps t-steps--horizontal t-steps--default-anchor t-steps--positive"
  >
    <div
      class="t-steps-item t-steps-item--process"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <svg
            class="t-icon t-icon-check-circle-filled"
            version="1.1"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM690.8655168 353.58818368L464.56401024 579.82818368 328.82684416 444.09785152 283.57201024 489.35268544 464.59134592 670.3720192 509.84617984 625.1171872 736.120352 398.8430176 690.8655168 353.58818368Z"
              fill-opacity=".9"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            已完成的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <svg
            class="t-icon t-icon-check-circle-filled"
            version="1.1"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM690.8655168 353.58818368L464.56401024 579.82818368 328.82684416 444.09785152 283.57201024 489.35268544 464.59134592 670.3720192 509.84617984 625.1171872 736.120352 398.8430176 690.8655168 353.58818368Z"
              fill-opacity=".9"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            进行中的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <svg
            class="t-icon t-icon-check-circle-filled"
            version="1.1"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM690.8655168 353.58818368L464.56401024 579.82818368 328.82684416 444.09785152 283.57201024 489.35268544 464.59134592 670.3720192 509.84617984 625.1171872 736.120352 398.8430176 690.8655168 353.58818368Z"
              fill-opacity=".9"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            未进行的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <svg
            class="t-icon t-icon-check-circle-filled"
            version="1.1"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM690.8655168 353.58818368L464.56401024 579.82818368 328.82684416 444.09785152 283.57201024 489.35268544 464.59134592 670.3720192 509.84617984 625.1171872 736.120352 398.8430176 690.8655168 353.58818368Z"
              fill-opacity=".9"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            未进行的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`no-sequence.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-steps t-steps--horizontal t-steps--positive t-steps--dot-anchor"
  >
    <div
      class="t-steps-item t-steps-item--finish"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        />
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            已完成的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--process"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        />
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            进行中的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        />
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            未进行的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        />
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            未进行的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`sequence.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-steps t-steps--horizontal t-steps--default-anchor t-steps--positive"
  >
    <div
      class="t-steps-item t-steps-item--finish"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <svg
            class="t-icon t-icon-check-circle-filled"
            version="1.1"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM690.8655168 353.58818368L464.56401024 579.82818368 328.82684416 444.09785152 283.57201024 489.35268544 464.59134592 670.3720192 509.84617984 625.1171872 736.120352 398.8430176 690.8655168 353.58818368Z"
              fill-opacity=".9"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            已完成的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--process"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            2
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            进行中的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            3
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            未进行的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            4
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            未进行的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`status.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-steps t-steps--horizontal t-steps--default-anchor t-steps--positive"
  >
    <div
      class="t-steps-item t-steps-item--finish"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <svg
            class="t-icon t-icon-check-circle-filled"
            version="1.1"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM690.8655168 353.58818368L464.56401024 579.82818368 328.82684416 444.09785152 283.57201024 489.35268544 464.59134592 670.3720192 509.84617984 625.1171872 736.120352 398.8430176 690.8655168 353.58818368Z"
              fill-opacity=".9"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            已完成的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--process"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            2
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            进行中的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--error"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <svg
            class="t-icon t-icon-close-circle"
            version="1.1"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM512 160C317.59576832 160 160 317.59576832 160 512 160 706.4042304 317.59576832 864 512 864 706.4042304 864 864 706.4042304 864 512 864 317.59576832 706.4042304 160 512 160ZM647.7644992 330.98066432L693.0193344 376.23549824 557.22066432 511.97266432 693.0193344 647.7644992 647.7644992 693.0193344 511.97266432 557.22066432 376.23549824 693.0193344 330.98066432 647.7644992 466.72466432 511.97266432 330.98066432 376.23549824 376.23549824 330.98066432 511.97266432 466.72466432 647.7644992 330.98066432Z"
              fill-opacity=".9"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            错误的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            4
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            未进行的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`vertical-no-sequence.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-steps t-steps--vertical t-steps--default-anchor t-steps--positive"
  >
    <div
      class="t-steps-item t-steps-item--finish"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <svg
            class="t-icon t-icon-check-circle-filled"
            version="1.1"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM690.8655168 353.58818368L464.56401024 579.82818368 328.82684416 444.09785152 283.57201024 489.35268544 464.59134592 670.3720192 509.84617984 625.1171872 736.120352 398.8430176 690.8655168 353.58818368Z"
              fill-opacity=".9"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            已完成的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--process"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            2
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            进行中的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            3
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            未进行的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            4
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            未进行的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`vertical-sequence.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-steps t-steps--vertical t-steps--default-anchor t-steps--positive"
  >
    <div
      class="t-steps-item t-steps-item--finish"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <svg
            class="t-icon t-icon-check-circle-filled"
            version="1.1"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM690.8655168 353.58818368L464.56401024 579.82818368 328.82684416 444.09785152 283.57201024 489.35268544 464.59134592 670.3720192 509.84617984 625.1171872 736.120352 398.8430176 690.8655168 353.58818368Z"
              fill-opacity=".9"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            已完成的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--process"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            2
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            进行中的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            3
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            未进行的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-steps-item t-steps-item--wait"
    >
      <div
        class="t-steps-item__inner"
      >
        <div
          class="t-steps-item-icon"
        >
          <span
            class="t-steps-item-icon__number"
          >
            4
          </span>
        </div>
        <div
          class="t-steps-item-content"
        >
          <div
            class="t-steps-item-title"
          >
            未进行的步骤
          </div>
          <div
            class="t-steps-item-description"
          >
            这里是提示文字
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
